src/Bicep.Core.IntegrationTests/EvaluationTests.cs

```diff
@@ -976,5 +981,53 @@ public void Safe_dereferences_are_evaluated_successfully()
                 evaluated.Should().HaveValueAtPath("$.outputs['properties'].value.doesntExistArrayAccess", JValue.CreateNull());
             }
         }
+
+        [TestMethod]
+        public void Assertions_are_evaluated_correctly()
+        {
+            var services = new ServiceBuilder().WithFeatureOverrides(new(TestContext, AssertsEnabled: true));
+            var bicepFile = @"
+param accountName string
+param environment string
+param location string
+
+resource stgAccount 'Microsoft.Storage/storageAccounts@2019-06-01' = {
+  name: toLower(accountName)
+  location: resourceGroup().location
+  kind: 'Storage'
+  sku: {
+    name: 'Standard_LRS'
+  }
+}
+
+var myInt = 24
+
+assert a1 = length(accountName) < myInt
+assert a2 = contains(location, 'us')
+assert a3 = environment == 'dev'
+";
+
+
+            var paramsFile = @"
+using 'main.bicep'
+
+// long string to trigger the assertion
+param accountName = 'asdgkbauskfabdsfibasdogbnasdognbaosdingoaisdngoisdangoinbdsaoigbsadoibgodsiabgos'
+param environment = 'dev'
+param location = 'westus'
+";
+
+            var (parameters, _, _) = CompilationHelper.CompileParams(services, ("parameters.bicepparam", paramsFile), ("main.bicep", bicepFile));
+            var (template, _, _) = CompilationHelper.Compile(services, bicepFile);
+
+            using (new AssertionScope())
+            {
+                var evaluated = TemplateEvaluator.Evaluate(template, parameters);
+                
+                evaluated.Should().HaveValueAtPath("$.asserts['a1']", false);
```

Comment:
Not germane to this PR, but should we maybe emit a warning diagnostic when it can be determined at compile time that an assertion will be false?